### PR TITLE
Add docs for GetRestApi

### PR DESCRIPTION
**Pull request by jogonzal**
Adding docs for GetRestApi function in OneNote pages, sections, sectionGroups, and notebooks

### CODE REVIEW

File: reference/onenote/notebook.md [hunk 1]
```diff
@@ -119,6 +120,35 @@ OneNote.run(function (context) {
...
+#### Returns
+string
+
+#### Examples
+```js
+
+OneNote.run(function(ctx){
+    // Get the current notebook.         
+    var notebook = ctx.application.getActiveNotebook();
+    var restApiId = notebook.getRestApiId();
```
Comment (hanjung) [inline reference/onenote/notebook.md]: can you add a line for using baseUrl property for SharePoint? add comment to say needed for SharePoint and perhaps a link to our REST API doc?

Comment (hanjung) [inline reference/onenote/notebook.md]: You also need to expose notebook.baseUrl and have similar comments

Comment (jogonzal) [inline reference/onenote/notebook.md]: You're right, will add this.

Comment (jogonzal) [inline reference/onenote/notebook.md]: @hanjung I've made these updates - can you take a look?

File: reference/onenote/notebook.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ Represents a OneNote notebook. Notebooks contain section groups and sections.
 
 | Property	   | Type	|Description|Feedback|
 |:---------------|:--------|:----------|:-------|
+|baseUrl|string|The base site url for the notebook, if it is in a SharePoint site (it will be null for OneDrive notebooks) - this is to be used for interacting with the OneNote REST API, according to our [OneNote API documentation](https://blogs.msdn.microsoft.com/onenotedev/2015/06/11/and-sharepoint-makes-three/) to fetch the SiteCollectionId and SiteId (using FromUrl). Read only.|[Go](https://github.com/OfficeDev/office-js-docs/issues/new?title=OneNote-notebook-baseUrl)|
 |clientUrl|string|The client url of the notebook. Read only Read-only.|[Go](https://github.com/OfficeDev/office-js-docs/issues/new?title=OneNote-notebook-clientUrl)|
```
Comment (Lauragra) [inline reference/onenote/notebook.md]: Please remove the dupe "Read only" here.

File: reference/onenote/notebook.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ Represents a OneNote notebook. Notebooks contain section groups and sections.
 
 | Property	   | Type	|Description|Feedback|
 |:---------------|:--------|:----------|:-------|
+|baseUrl|string|The base site url for the notebook, if it is in a SharePoint site (it will be null for OneDrive notebooks) - this is to be used for interacting with the OneNote REST API, according to our [OneNote API documentation](https://blogs.msdn.microsoft.com/onenotedev/2015/06/11/and-sharepoint-makes-three/) to fetch the SiteCollectionId and SiteId (using FromUrl). Read only.|[Go](https://github.com/OfficeDev/office-js-docs/issues/new?title=OneNote-notebook-baseUrl)|
```
Comment (Lauragra) [inline reference/onenote/notebook.md]: Change url to URL.

Comment (Lauragra) [inline reference/onenote/notebook.md]: Remove the hyphen and create a new sentence as follows: "Use this property to interact with the OneNote REST API to fetch the **SiteCollectionId** and **SiteId** (using **FromUrl**). Read-only. For details, see the [OneNote development blog](https://blogs.msdn.microsoft.com/onenotedev/2015/06/11/and-sharepoint-makes-three/).

Note that programming elements should have bold formatting applied.

File: reference/onenote/notebook.md [hunk 4]
```diff
@@ -27,6 +28,7 @@ _See property access [examples.](#property-access-examples)_
 |:---------------|:--------|:----------|:-------|
 |[addSection(name: String)](#addsectionname-string)|[Section](section.md)|Adds a new section to the end of the notebook.|[Go](https://github.com/OfficeDev/office-js-docs/issues/new?title=OneNote-notebook-addSection)|
 |[addSectionGroup(name: String)](#addsectiongroupname-string)|[SectionGroup](sectiongroup.md)|Adds a new section group to the end of the notebook.|[Go](https://github.com/OfficeDev/office-js-docs/issues/new?title=OneNote-notebook-addSectionGroup)|
+|[getRestApiId()](#getRestApiId)|string|Gets the id that is compatible with the REST API.|[Go](https://github.com/OfficeDev/office-js-docs/issues/new?title=OneNote-notebook-getRestApiId)|
```
Comment (Lauragra) [inline reference/onenote/notebook.md]: Let's cap "ID" here.

File: reference/onenote/notebook.md [hunk 5]
```diff
@@ -119,6 +121,37 @@ OneNote.run(function (context) {
...
+
+OneNote.run(function(ctx){
+    // Get the current notebook.         
+    var notebook = ctx.application.getActiveNotebook();
+    var restApiId = notebook.getRestApiId();
+
+    return ctx.sync().
+        then(function(){
+            console.log("The rest api id is " + restApiId.value);
+            // Note that the rest API id isn't all you need to interact with the OneNote Rest API. For SharePoint notebooks, the notebook baseUrl should be used to talk to OneNote REST APIs according to https://blogs.msdn.microsoft.com/onenotedev/2015/06/11/and-sharepoint-makes-three/
```
Comment (Lauragra) [inline reference/onenote/notebook.md]: Change "rest API id" to "REST API ID", and change "OneNote Rest API" to "OneNote REST API". Also, change the URL to a link - [OneNote development blog](https://blogs.msdn.microsoft.com/onenotedev/2015/06/11/and-sharepoint-makes-three/). (The title of that blog post isn't super clear that it applies to OneNote.)

File: reference/onenote/notebook.md [hunk 6]
```diff
@@ -135,6 +168,32 @@ object.load(param);
...
+            
+    // Queue a command to load the notebook. 
+    // For best performance, request specific properties.           
+    notebook.load('baseUrl');
+            
+    // Run the queued commands, and return a promise to indicate task completion.
+    return context.sync()
+        .then(function () {
+            console.log("Base url: " + notebook.baseUrl);
+            // This baseUrl should be used to talk to OneNote REST APIs according to https://blogs.msdn.microsoft.com/onenotedev/2015/06/11/and-sharepoint-makes-three/ (only required for SharePoint notebooks, will be null for OneDrive notebooks)
```
Comment (Lauragra) [inline reference/onenote/notebook.md]: Change the blog post link here as well.

File: reference/onenote/page.md [hunk 7]
```diff
@@ -30,6 +30,7 @@ _See property access [examples.](#property-access-examples)_
 |:---------------|:--------|:----------|:-------|
 |[addOutline(left: double, top: double, html: String)](#addoutlineleft-double-top-double-html-string)|[Outline](outline.md)|Adds an Outline to the page at the specified position.|[Go](https://github.com/OfficeDev/office-js-docs/issues/new?title=OneNote-page-addOutline)|
 |[copyToSection(destinationSection: Section)](#copytosectiondestinationsection-section)|[Page](page.md)|Copies this page to specified section.|[Go](https://github.com/OfficeDev/office-js-docs/issues/new?title=OneNote-page-copyToSection)|
+|[getRestApiId()](#getRestApiId)|string|Gets the id that is compatible with the REST API.|[Go](https://github.com/OfficeDev/office-js-docs/issues/new?title=OneNote-page-getRestApiId)|
```
Comment (Lauragra) [inline reference/onenote/page.md]: Change to ID.

File: reference/onenote/page.md [hunk 8]
```diff
@@ -146,6 +147,37 @@ OneNote.run(function(ctx) {
 });
 ```
 
+### getRestApiId()
+Gets the id that is compatible with the REST API.
```
Comment (Lauragra) [inline reference/onenote/page.md]: ID.

File: reference/onenote/page.md [hunk 9]
```diff
@@ -146,6 +147,37 @@ OneNote.run(function(ctx) {
...
+
+OneNote.run(function(ctx){
+    // Get the current page.         
+    var page = ctx.application.getActivePage();
+    var restApiId = page.getRestApiId();
+
+    return ctx.sync().
+        then(function(){
+            console.log("The rest api id is " + restApiId.value);
+            // Note that the rest API id isn't all you need to interact with the OneNote Rest API. For SharePoint notebooks, the notebook baseUrl should be used to talk to OneNote REST APIs according to https://blogs.msdn.microsoft.com/onenotedev/2015/06/11/and-sharepoint-makes-three/
```
Comment (Lauragra) [inline reference/onenote/page.md]: "REST API ID", "OneNote REST API", fix the link here too.

File: reference/onenote/section.md [hunk 10]
```diff
@@ -31,6 +31,7 @@ _See property access [examples.](#property-access-examples)_
 |[copyToNotebook(destinationNotebook: Notebook)](#copytonotebookdestinationnotebook-notebook)|[Section](section.md)|Copies this section to specified notebook.|[Go](https://github.com/OfficeDev/office-js-docs/issues/new?title=OneNote-section-copyToNotebook)|
 |[copyToSectionGroup(destinationSectionGroup: SectionGroup)](#copytosectiongroupdestinationsectiongroup-sectiongroup)|[Section](section.md)|Copies this section to specified section group.|[Go](https://github.com/OfficeDev/office-js-docs/issues/new?title=OneNote-section-copyToSectionGroup)|
 |[insertSectionAsSibling(location: string, title: string)](#insertsectionassiblinglocation-string-title-string)|[Section](section.md)|Inserts a new section before or after the current section.|[Go](https://github.com/OfficeDev/office-js-docs/issues/new?title=OneNote-section-insertSectionAsSibling)|
+|[getRestApiId()](#getRestApiId)|string|Gets the id that is compatible with the REST API.|[Go](https://github.com/OfficeDev/office-js-docs/issues/new?title=OneNote-section-getRestApiId)|
```
Comment (Lauragra) [inline reference/onenote/section.md]: ID.

File: reference/onenote/section.md [hunk 11]
```diff
@@ -224,6 +225,37 @@ OneNote.run(function (context) {
 });
 ```
 
+### getRestApiId()
+Gets the id that is compatible with the REST API.
```
Comment (Lauragra) [inline reference/onenote/section.md]: ID.

File: reference/onenote/section.md [hunk 12]
```diff
@@ -224,6 +225,37 @@ OneNote.run(function (context) {
...
+
+OneNote.run(function(ctx){
+    // Get the current section.         
+    var section = ctx.application.getActiveSection();
+    var restApiId = section.getRestApiId();
+
+    return ctx.sync().
+        then(function(){
+            console.log("The rest api id is " + restApiId.value);
+            // Note that the rest API id isn't all you need to interact with the OneNote Rest API. For SharePoint notebooks, the notebook baseUrl should be used to talk to OneNote REST APIs according to https://blogs.msdn.microsoft.com/onenotedev/2015/06/11/and-sharepoint-makes-three/
```
Comment (Lauragra) [inline reference/onenote/section.md]: Same changes here - REST API, ID, add the link.

File: reference/onenote/sectiongroup.md [hunk 13]
```diff
@@ -30,6 +30,7 @@ _See property access [examples.](#property-access-examples)_
 |:---------------|:--------|:----------|:-------|
 |[addSection(title: String)](#addsectiontitle-string)|[Section](section.md)|Adds a new section to the end of the section group.|[Go](https://github.com/OfficeDev/office-js-docs/issues/new?title=OneNote-sectionGroup-addSection)|
 |[addSectionGroup(name: String)](#addsectiongroupname-string)|[SectionGroup](sectiongroup.md)|Adds a new section group to the end of this sectionGroup.|[Go](https://github.com/OfficeDev/office-js-docs/issues/new?title=OneNote-sectionGroup-addSectionGroup)|
+|[getRestApiId()](#getRestApiId)|string|Gets the id that is compatible with the REST API.|[Go](https://github.com/OfficeDev/office-js-docs/issues/new?title=OneNote-sectionGroup-getRestApiId)|
```
Comment (Lauragra) [inline reference/onenote/sectiongroup.md]: ID

**Comment by Lauragra** [comment]
Looks good - thanks!

**Comment by jogonzal** [comment]
Thanks!

**Pull request merged**